File: python/hamming/hamming.py
def distance(first, second):

    hamdist = 0
    zipped = zip(first, second)

    if len(first) != len(second):
        raise ValueError("Strands must be of equal length.")

    for (x, y) in zipped:
       if x != y:
           hamdist += 1

    return hamdist

File: typescript/hamming/hamming.ts
export function compute(left: string, right: string): number {
  if (left.length !== right.length) {
    throw new Error('DNA strands must be of equal length.');
  }

  let hamming = 0;
  for (let i = 0; i < left.length; i++) {
    if (left[i] !== right[i]) hamming++;
  }
  return hamming;
}
